src/Compilers/CSharp/Portable/Symbols/SymbolWithAnnotations.cs

```diff
@@ -580,6 +580,17 @@ public TypeSymbolWithAnnotations SetUnknownNullabilityForReferenceTypesIfNecessa
                 this.SetUnknownNullabilityForReferenceTypes();
         }
 
+        public TypeSymbolWithAnnotations WithTopLevelNonNullability()
+        {
+            var typeSymbol = TypeSymbol;
+            if (IsNullable == false || typeSymbol.IsNullableType() || !typeSymbol.IsReferenceType)
```

Comment:
>typeSymbol.IsNullableType() [](start = 39, length = 27)

I don't think the `typeSymbol.IsNullableType()` test is needed, because it always implies `!typeSymbol.IsReferenceType`.